app/services/base_index_job.rb

```diff
@@ -21,6 +21,7 @@ def initialize(book_version_id: nil,
 
   def call
     _call
+  ensure #it should always remove the job from the queue
```

Comment:
Can you exercise this change in a spec, even with some stubbing in a dummy job where the `cleanup_after_call` proc is a spy that you can assert receives `:call`?